src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -300,30 +300,30 @@ std::unique_ptr<VideoDecoder> VideoDecoder::createFromBuffer(
 }
 
 void VideoDecoder::initializeFilterGraphForStream(
-    int streamIndex,
-    const VideoStreamDecoderOptions& options) {
-  FilterState& filterState = streams_[streamIndex].filterState;
+    StreamInfo& streamInfo,
+    const AVFrame& frame) {
```

Comment:
(Not critical for merging)

Instead of passing the entire AVFrame object, should we just pass the height and width? It would make it more obvious what properties from the AVFrame are really needed for initialization, and we could directly pass `expectedOutputHeight` and `expectedOutputWidth` from within `convertAVFrameToDecodedOutputOnCPU`?

Reply:
I thought about that too, and decided against it so I could use the `getHeightAndWidthFromOptionsOrAVFrame()` convenience function. But let me take another look at the code, because I agree that I don't like passing around the AVFrame pointer more than we have to.

Reply:
Oh, duh, there's no need to even call the convenience function inside `getHeightAndWidthFromOptionsOrAVFrame()`. We call it outside, pass the values in, no need to check options again.